internal/provider/cluster_resource.go

```diff
@@ -404,6 +411,12 @@ func (r *clusterResource) Create(
 		clusterSpec.SetParentId(parentID)
 	}
 
+	deleteProtection := client.DELETEPROTECTIONSTATETYPE_DISABLED
+	if plan.DeleteProtection.ValueBool() {
+		deleteProtection = client.DELETEPROTECTIONSTATETYPE_ENABLED
+	}
+	clusterSpec.SetDeleteProtection(deleteProtection)
```

Comment:
are the client constants actually just bools?  Could we just:

```go
clusterSpec.SetDeleteProtection(plan.DeleteProtection.ValueBool())
```

Reply:
No the API uses enums so I have to map from bool to the enum value.